backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,68 @@
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import Collection, CollectionMetadata, CollectionVersion, DatasetAsset, DatasetMetadata, DatasetStatus
+
+
+class DatabaseProviderInterface:
+    def create_collection(self, collection_metadata: CollectionMetadata):
+        pass
+
+    def get_collection(self, collection_id: str) -> Collection:
+        pass
+
+    def get_all_collections(self) -> List[Collection]:  # TODO: add filters if needed
+        pass
+
+    def delete_collection(self, collection_id: str) -> None:
+        pass
+
+    def update_collection_metadata(self, version_id: str, collection_metadata: CollectionMetadata) -> None:
+        pass
+
+    def update_collection_publisher_metadata(self, version_id: str, publisher_metadata: dict) -> None:
+        pass
+
+    def create_collection_version(self, collection_id: str) -> str:
+        pass
+
+    def delete_collection_version(self, collection_id: str, version_id: str) -> None:
+        pass
+
+    def get_collection_version(self, collection_id: str, version_id: str) -> CollectionVersion:
```

Comment:
if we end up adding `get_latest_collection_version`, we may not need this.  i.e., will we ever need to get a collection version that isn't the latest one? (not sure, just thinking out loud).   If anything, we might need a `get_all_collection_versions(self, collection_id:str)`, if we have business logic that needs to work with more than just the latest version.

Reply:
I thought it might be safer not to rely on `get_all_collection_versions`, but to have the API pass the version_id it's currently working on. This can avoid some race conditions. We might also not follow this pattern, and in that case, we won't need this method.